net/libfab/libfab.c

```diff
@@ -545,22 +551,55 @@ static void libfab_txep_event_check(struct m0_fab__ep *txep,
 }
 
 /**
- * Check for completion events on the for the rx ep
+ * Check for completion events on the CQ for the rx ep
  */
 static void libfab_rxep_comp_read(struct fid_cq *cq, struct m0_fab__ep *ep)
 {
-	struct m0_fab__buf *buf = NULL;
-	m0_bindex_t         len = 0;
+	struct m0_fab__buf *buf[FAB_MAX_COMP_READ];
+	m0_bindex_t         len[FAB_MAX_COMP_READ];
+	int                 i;
 	int                 rc;
 
+	memset(&buf, 0, sizeof(buf));
+	memset(&len, 0, sizeof(len));
 	if (cq != NULL) {
-		rc = libfab_check_for_comp(cq, &buf, &len);
+		rc = libfab_check_for_comp(cq, buf, len);
 		if (rc > 0) {
-			if (buf->fb_length == 0)
-				buf->fb_length = len;
-			buf->fb_ev_ep = ep;
-			libfab_ep_get(ep);
-			libfab_buf_done(buf, 0);
+			for (i = 0; i < rc; i++) {
```

Comment:
The condition "if (rc > 0)" is covered by "for (i = 0; i < rc; i++)".
Can we drop the "if (rc > 0"?

Reply:
Done